apps/yapms/src/lib/components/modals/toolsmodal/tools/TableModal.svelte

```diff
@@ -0,0 +1,128 @@
+<script lang="ts">
+	import { TableModalStore } from '$lib/stores/Modals';
+	import { RegionsStore } from '$lib/stores/regions/Regions';
+	import type { Region } from '$lib/types/Region';
+	import ModalBase from '../../ModalBase.svelte';
+	import { disableRegion, lockRegion } from '$lib/stores/regions/regionActions';
+	import { get } from 'svelte/store';
+	import CandidateActions from './tablecomponents/CandidateActions.svelte';
+
+	let filterInput = '';
+	$: isFiltering = filterInput !== '';
+	$: regionsToShow = filterRegions(filterInput);
```

Comment:
optional nit-pick:

If you reduce the regionsToShow to this line of code, you can eliminate a lot of your logic. It will just create the sorted and filtered list of regions. And will automatically update as you type.

![image](https://github.com/user-attachments/assets/c83d92b1-4cee-417b-9fe5-f8640440505f)

Reply:
This is still performant, so implemented.